administration/src/bp-modules/cards/ImportCardsFromFreinetController.ts

```diff
@@ -0,0 +1,66 @@
+import { BAVARIA_CARD_TYPE_GOLD, BAVARIA_CARD_TYPE_STANDARD } from '../../cards/extensions/BavariaCardTypeExtension'
+import { ProjectConfig } from '../../project-configs/getProjectConfig'
+
+const FREINET_EXPIRATION_DATE_COLUMN_HEADER = 'eak_datum'
+const FREINET_FIRSTNAME_COLUMN_HEADER = 'vorname'
+const FREINET_LASTNAME_COLUMN_HEADER = 'nachname'
+const FREINET_CARDTYPE_COLUMN_HEADER = 'inhaber_ehrenamtskarte'
+
+const mergeFirstAndLastnameIntoNewColumn = (line: string[], csvHeader: string[], nameColumnName: string) => {
+  const columnNameDoesNotExist = csvHeader.indexOf(nameColumnName) == -1
+  if (columnNameDoesNotExist) {
+    csvHeader[csvHeader.length] = nameColumnName
+  }
+
+  const indexVorname = csvHeader.indexOf(FREINET_FIRSTNAME_COLUMN_HEADER)
+  const indexNachname = csvHeader.indexOf(FREINET_LASTNAME_COLUMN_HEADER)
+  const fullname = `${line[indexVorname] ?? ''} ${line[indexNachname] ?? ''}`
+  line[csvHeader.indexOf(nameColumnName)] = fullname.trim()
+}
+
+const renameExpirationDateHeader = (csvHeader: string[], expiryColumnName: string) => {
+  const indexExpirationDate = csvHeader.indexOf(FREINET_EXPIRATION_DATE_COLUMN_HEADER)
+  if (indexExpirationDate != -1) {
+    csvHeader[indexExpirationDate] = expiryColumnName
+  }
+}
+
+const getCardTypeByFreinetValue = (line: string[], indexCardTypeFreinet: number): string =>
+  line[indexCardTypeFreinet].toLowerCase().includes('blau') ? BAVARIA_CARD_TYPE_STANDARD : BAVARIA_CARD_TYPE_GOLD
+
+const getCardTypeByExpirationDate = (line: string[], indexOfExpirationDate: number): string => {
+  return !!line[indexOfExpirationDate] ? BAVARIA_CARD_TYPE_STANDARD : BAVARIA_CARD_TYPE_GOLD
+}
+
+const setCardType = (line: string[], csvHeader: string[], cardTypeColumnName: string, expiryColumnName: string) => {
+  const indexCardTypeFreinet = csvHeader.indexOf(FREINET_CARDTYPE_COLUMN_HEADER)
+
+  const hasValidCardTypeColumn = csvHeader.indexOf(cardTypeColumnName) != -1
+  if (!hasValidCardTypeColumn) {
+    csvHeader[csvHeader.length] = cardTypeColumnName
+  }
+  const indexCardType = csvHeader.indexOf(cardTypeColumnName)
+  const hasFreinetCardType = indexCardTypeFreinet != -1
+
+  line[indexCardType] = hasFreinetCardType
+    ? getCardTypeByFreinetValue(line, indexCardTypeFreinet)
+    : (line[indexCardTypeFreinet] = getCardTypeByExpirationDate(line, csvHeader.indexOf(expiryColumnName)))
```

Comment:
:wrench: An assignment as else value in a ternary looks very weird. I think this would be better:
```suggestion
if (hasFreinetCardType) {
  line[indexCardType] = getCardTypeByFreinetValue(line, indexCardTypeFreinet)
} else {
  line[indexCardTypeFreinet] = getCardTypeByExpirationDate(line, csvHeader.indexOf(expiryColumnName))
}
```
Or am I missing something here?

Reply:
Thanks you are right. I fixed this.